controller/cmd/service-mirror/metrics.go

```diff
@@ -0,0 +1,73 @@
+package servicemirror
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var labelNames = []string{
+	"gateway_name", "gateway_namespace", "remote_cluster_name",
+}
+
+type probeMetricVecs struct {
+	services  *prometheus.GaugeVec
+	alive     *prometheus.GaugeVec
+	latencies *prometheus.HistogramVec
+}
+
+type probeMetrics struct {
+	services  prometheus.Gauge
+	alive     prometheus.Gauge
```

Comment:
in addition to having a gauge for "alive", we could also have a counter for "probes" with a label to indicate if the probe succeeded or failed.

Reply:
I think a counter only goes up, right? And the number of probes is sublject to going down as well. What is the advantage of having this counter in addition the gauge. Does it give us something more? If we grab the gauges, we ultimatelly get the number of probes along with the information of whether the gateway is alive or not. 

Or  are you trying to measure the total number of attemptes to reach a remote cluster and partition them by success outcome?

Reply:
Yeah, that's right.  A total number of attempts, rather than a gauge of in-flight attempts.  This allows us to see the rate of probes over time.  For example, if for some reason probes stop happening, the alive gauge would stay at `1` and we would not be able to see that probes had stopped.  A counter would let us see this.

Reply:
This is done